addons/account/models/partner.py

```diff
@@ -514,6 +514,17 @@ def create(self, vals_list):
                     vals['supplier_rank'] = 1
         return super().create(vals_list)
 
+    def unlink(self):
+        """
+        Prevent the deletion of a partner "Individual", child of a company if:
+        - partner in 'account.move'
+        - state: all states (draft and posted)
+        """
+        moves = self.env['account.move'].search([('partner_id', '=', self.id), ('state', '=', 'posted')])
```

Comment:
use sudo() to check on all aml (if user have no access) and search_count to prevent orm return all aml.